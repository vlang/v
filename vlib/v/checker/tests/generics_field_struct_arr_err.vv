// callback types
type CBnoret[T] = fn (val T)

type CBnoret2[T] = fn (val T, prev T)

type CBvret[T] = fn (val T) T

type CBvret2[T] = fn (val T, prev T) T

type Callback[T] = CBnoret[T] | CBnoret2[T] | CBvret[T] | CBvret2[T]

interface IObv[T] {
	v    T
	prev T
	cb   []Callback[T]
}

struct Test[T] {
	a T
}

struct Obv[T] {
mut:
	v    T
	prev T
	cb   []Callback // sumtype
	cb2  []CBvret2  // alias
	cb3  []Test     // struct
}
