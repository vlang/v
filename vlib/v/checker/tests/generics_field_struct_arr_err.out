vlib/v/checker/tests/generics_field_struct_arr_err.vv:26:7: error: `Callback` type is generic sumtype, must specify the generic type names, e.g. Callback[T], Callback[int]
   24 |     v    T
   25 |     prev T
   26 |     cb   []Callback // sumtype
      |          ~~~~~~~~~~
   27 |     cb2  []CBvret2  // alias
   28 |     cb3  []Test     // struct
vlib/v/checker/tests/generics_field_struct_arr_err.vv:27:7: error: `CBvret2` type is generic fn type, must specify the generic type names, e.g. CBvret2[T], CBvret2[int]
   25 |     prev T
   26 |     cb   []Callback // sumtype
   27 |     cb2  []CBvret2  // alias
      |          ~~~~~~~~~
   28 |     cb3  []Test     // struct
   29 | }
vlib/v/checker/tests/generics_field_struct_arr_err.vv:28:7: error: `Test` type is generic struct, must specify the generic type names, e.g. Test[T], Test[int]
   26 |     cb   []Callback // sumtype
   27 |     cb2  []CBvret2  // alias
   28 |     cb3  []Test     // struct
      |          ~~~~~~
   29 | }
